src/pages/home/index.js

```diff
@@ -0,0 +1,582 @@
+
+import { useClasses } from "../../lib";
+import VaccineDataMain from "../../components/VaccineData/VaccineDataMain";
+import Pagination from "../../components/Pagination/Pagination";
+
+/**********************************************************/
+/********************** NEEDS FIXING **********************/
+/**********************************************************/
+
+// import NullState from "../../components/NullState";
+// import Error from "../../components/Error";
+
+/**********************************************************/
+
+const BASE_URL = "https://cdn-api.co-vin.in/api/v2";
+
+export const HomePage = () => {
+  const classes = useClasses();
+
+  /** ASYNC **/
+  const [loadingStates, setLoadingStates] = React.useState(false);
+  const [loadingDistricts, setLoadingDistricts] = React.useState(false);
+  const [error, setError] = React.useState(null);
+
+  /** CRITERIA **/
+  const [selectedCriteria, setSelectedCriteria] = React.useState(null);
+
+  /** STATES **/
+  const [states, setStates] = React.useState(null);
+  const [selectedState, setSelectedState] = React.useState(null);
+
+  /** DISTRICTS **/
+  const [districts, setDistricts] = React.useState(null);
+  const [selectedDistrict, setSelectedDistrict] = React.useState(null);
+
+  /** PIN **/
+  const [selectedPin, setSelectedPin] = React.useState(null);
+  const [invalidPin, setInvalidPin] = React.useState(false);
+
+  /** VACCINES **/
+  const [vaccines, setVaccines] = React.useState(null);
+  const [currentVaccines, setCurrentVaccines] = React.useState(null);
+  const [loadingVaccines, setLoadingVaccines] = React.useState(false);
+
+  /** MISC **/
+  const [date, setDate] = React.useState(new Date());
+
+  /** PAGINATION **/
+  const [currentPage, setCurrentPage] = React.useState(1);
+  const [vaccinePerPage] = React.useState(3);
+  const indexOfLastVaccine = currentPage * vaccinePerPage;
+  const indexOfFirstVaccine = indexOfLastVaccine - vaccinePerPage;
+  const [pages, setPages] = React.useState([]);
+
+  React.useEffect(() => {
+    if (vaccines && vaccines.length) {
+      setCurrentVaccines(
+        vaccines.slice(indexOfFirstVaccine, indexOfLastVaccine)
+      );
+      for (let i = 1; i <= Math.ceil(vaccines.length / vaccinePerPage); i++) {
+        setPages((old) => [...old, i]);
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [vaccines, currentPage]);
+
+  const paginate = (pageNumber) => {
+    setCurrentPage(pageNumber);
+  };
+
+  const nextPage = () => {
+    setCurrentPage(currentPage + 1);
+
+    if (currentPage + 1 > pages.length) {
+      setCurrentPage(pages.length);
+    }
+  };
+
+  const prevPage = () => {
+    setCurrentPage(currentPage - 1);
+
+    if (currentPage - 1 <= 0 && pages.length) {
+      setCurrentPage(pages.slice(0, 1));
+    }
+  };
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  React.useEffect(() => {
+    if (!states) {
+      if (/district/gi.test(selectedCriteria)) {
```

Comment:
Also, wanted to understand what exactly this condition is checking, i understood this is a regex, but was not understand the use of ".test" 
is checking if a district is being selected from the dropdown? 🤔 then call getStates() function.


I would love to understand this type of checks, looks cool though 😎

Reply:
it's checking if the value of the `selectedCriteria` variable matches the regex `/disctrict/gi` ( "Find by district" or "Find by district & date" )